src/phpDocumentor/Transformer/Router/Router.php

```diff
@@ -39,86 +36,82 @@
  */
 class Router extends ArrayObject
 {
-    private $projectDescriptorBuilder;
+    /** @var UrlGenerator\FqsenDescriptor */
     private $fqsenUrlGenerator;
+
+    /** @var QualifiedNameToUrlConverter */
     private $converter;
+
+    /** @var UrlGeneratorInterface */
     private $urlGenerator;
 
     public function __construct(
-        ProjectDescriptorBuilder $projectDescriptorBuilder,
         UrlGenerator\FqsenDescriptor $fqsenUrlGenerator,
         QualifiedNameToUrlConverter $converter,
         UrlGeneratorInterface $urlGenerator
     ) {
-        $this->projectDescriptorBuilder = $projectDescriptorBuilder;
         $this->fqsenUrlGenerator = $fqsenUrlGenerator;
         $this->converter = $converter;
         $this->urlGenerator = $urlGenerator;
 
         parent::__construct();
+
         $this->configure();
     }
 
     /**
      * Configuration function to add routing rules to a router.
      */
-    public function configure()
+    public function configure() : void
     {
-        // Here we cheat! If a string element is passed to this rule then we try to transform it into a Descriptor
```

Comment:
Apparently, it was not necessary because of how the linker properly converts everything to an object and this was causing issues with type-hinting. I am still working on the routing and will probably reintroduce something that can route based on a string again in the future